analyzer/src/types/engine.rs

```diff
@@ -108,6 +108,10 @@ impl TypedEngine {
         }
     }
 
+    /// Returns the chunk with the given source id.
+    ///
+    /// If the chunk is not a user defined chunk, [`None`] is returned.
+    /// Use [`Self::get`] to get both user defined and native chunks.
     pub fn get_user(&self, id: SourceId) -> Option<&Chunk> {
```

Comment:
```suggestion
    pub fn get_user_defined_chunk(&self, id: SourceId) -> Option<&Chunk> {
```

Reply:
`Self::get()` already returns a `Chunk`, so this would not be in phase with this other method.